controllers/discordactions.js

```diff
@@ -63,6 +64,74 @@ const createGroupRole = async (req, res) => {
   }
 };
 
+/**
+ * Controller function to handle the soft deletion of a group role.
+ *
+ * @param {Object} req - The request object
+ * @param {Object} res - The response object
+ * @returns {Promise<void>}
+ */
+const deleteGroupRole = async (req, res) => {
```

Comment:
- Refactor this controller. Place everything in separate service.
- What will happen if the delete role from discord gets successful and db query fails?

Reply:
Can you explain on how to make a separate service. I've followed the same pattern as in the file

Reply:
- Controller should not be doing all these thing. It should accept request, call service, receive output and send response.
 - Write a function in discord-service file.

Reply:
There is already a function in discord service file which controller is calling

Reply:
We have deleteRoleFromDiscord function.
We can write another one deleteGroupRole right? and refactor controller.

Reply:
Not getting your point